### DISCUSSION TITLE
Limit connection count to vpn per user

### DISCUSSION THREAD

**Discussion by dotmitsu** [strongswan/strongswan#1083] [General]
Hi. I use strongswan server for multiple users connection to vpn. How can I limit max simultaneous number of connections per user to 3?
I need to give opportunity to connect simultaneous only 3 device to vpn for any user.
I can limit maximum connections count to server by decreasing subnet pool, but I have no idea how make limit for every user.
Please help me))

my ipsec.conf

```
config setup
    charondebug="ike 1, knl 1, cfg 0"
    uniqueids=never

conn ikev2-vpn
    auto=add
    compress=no
    type=tunnel
    keyexchange=ikev2
    fragmentation=yes
    forceencaps=yes
    dpdaction=clear
    dpddelay=300s
    rekey=no
    left=%any
    leftid=46.249.49.63
    leftcert=server-cert.pem
    leftsendcert=always
    leftsubnet=0.0.0.0/0
    right=%any
    rightid=%any
    rightauth=eap-mschapv2
    rightsourceip=172.27.27.0/24
    rightdns=8.8.8.8,8.8.4.4
    rightsendcert=never
```

my ipsec.secrets

```
: RSA "server-key.pem"

user1 : EAP "password"
user2 : EAP "password"
user3 : EAP "password"
```

**Comment by tobiasbrunner** [comment]
I guess the most flexible is using authentication via [RADIUS](https://docs.strongswan.org/docs/5.9/plugins/eap-radius.html). (Or maybe it could be done via RADIUS accounting and [DAE](https://docs.strongswan.org/docs/5.9/plugins/eap-radius.html#_session_timeout_and_dynamic_authorization_extension).)

User-specific IP pools might also work. But to switch connections based on the EAP identity you need [a workaround](https://serverfault.com/a/908194/95913).

The [ext-auth plugin](https://docs.strongswan.org/docs/5.9/plugins/ext-auth.html) could also be an option, but since it won't get called when a connection is terminated, it had to be combined with e.g. [vici events](https://docs.strongswan.org/docs/5.9/plugins/vici.html).
